### PR TITLE
Add SN2 Reaction Simulation

### PR DESCRIPTION
**Pull request by b30wulffz**

### CODE REVIEW

File: experiment/aim.md [hunk 1]
```diff
@@ -1 +1,5 @@
-### Aim of the experiment
+When two (especially organic) molecules approach each other, there is a possibility of several different chemical reactions between them; each giving different products. Many reactions occur in a single step if the reactants collide with a suffucient energy and specific orientation to form a product. However, many reactions do not occur in a single step but follows multi-steps where it goes through many intermediate levels in a half-way and finally leads to product. These intermediate levels are known as transition states. Stabilisation of these transition states is very difficult because these are highly energetic and unstable species. A chemical reaction may go through many transition states and ultimately leads to the product.The number of transition states may be different for different reaction path ways for a chemical reaction. There are many different reaction path ways which leads to products from reactants example: Electrophilic reactions, Nucleophilic reactions etc
```
Comment (raj-vlabs) [inline experiment/aim.md]: Multiple grammatical issues here:
1. 'collide with sufficient energy' in place of 'collide with a suffucient energy'.
2. Sufficient is also misspelled in the last line.
3. 'many reactions do not occur in a single step but follow multi-steps' in place of 'many reactions do not occur in a single step but follows multi-steps'.
4. Please run it through a grammar checker, too many to list.

File: experiment/experiment-name.md [hunk 2]
```diff
@@ -1 +1 @@
-## Experiment name
+## Transition State Calculation for a SN2 reaction
```
Comment (raj-vlabs) [inline experiment/experiment-name.md]: Do we need 'a' before SN2 here?

File: experiment/procedure.md [hunk 3]
```diff
@@ -1 +1,5 @@
-### Procedure
\ No newline at end of file
+1. An energy slider and structure of each molecule is present in the simulation's table.
+2. Adjust relative energy for each molecule present in Reactants, Transition State and Products, such that you obtain an ideal state vs energy plot.
+3. Note down these relative values, and compare it with actual values.
+4. Use mouse to interact with molecule, by rotating, and zooming around it.
```
Comment (raj-vlabs) [inline experiment/procedure.md]: Mouse will not be present on touch devices.

File: experiment/simulation/.vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (raj-vlabs) [inline experiment/simulation/.vscode/settings.json]: Please exclude this file from the repo.

File: experiment/simulation/js/main.js [hunk 5]
```diff
@@ -1 +1,135 @@
...
+};
+
+const chloride = "1\n \nCl  -0.0433700000  -0.0799600000   0.0000000000";
+generateMolecule("chloride-chart", chloride);
+
+const iodomethane =
+  "5\n \nI            0.3061173620   0.0001353896   0.0000861557\nC           -1.8735193700  -0.0000690343  -0.0000131229\nH           -2.2005433799   0.2316628203   0.9991454491\nH           -2.2003767307  -0.9812741146  -0.2989663895\nH           -2.2005235734   0.7493085882  -0.7003539349";
+generateMolecule("iodomethane-chart", iodomethane);
+
+const transitionstate =
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Please use camelCase naming for js variables and functions.

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Aim has multiple grammatical and spelling mistakes. Please run through a grammar validator.
2. Theory also needs to be reformatted and verified.
3. Looking at the simulation, I do not have any clue what I am supposed to do, esp with the input and output buttons on each item.
4. I am not clear why we are copying over ChemDoodleWeb files instead of linking with them. Copying means caching will not take effect.

File: experiment/posttest.json [hunk 6]
```diff
@@ -1,18 +1,76 @@
...
+        "d": "Propyl bromide"
+      },
+      "explanations": {
+        "c": "More groups on the C atom attached to bromide will make the attack of the nucleophile difficult and slow the reaction rate"
+      },
+      "correctAnswer": "c",
+      "difficulty": "beginner"
+    },
+    {
+      "question": "Which statement is true about S N 2 mechanism?",
```
Comment (raj-vlabs) [inline experiment/posttest.json]: There's gap between S N 2 at a few places in this file

File: experiment/simulation/index.html [hunk 7]
```diff
@@ -1,13 +1,255 @@
...
+            Note: All the energies here are relative. They do not account for
+            actual values.
+          </div>
+        </div>
+      </div>
+    </div>
+    <!-- Add JS at the bottom of HTML file -->
+    <script
+      type="text/javascript"
+      src="chemdoodleWeb-9.2.0/ChemDoodleWeb.js"
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Path needs to be updated to load it from the cdn.

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. There are a few errors on the simulation page, which should have been verified before submitting for the review. They are probably coming because the path to load ChemDoodleWeb has not been updated.
2. From the experiment it's still not clear what the user is supposed to do with the input and output buttons. This needs to be confirmed with Priya
3. On phone resolution, the input and output buttons start overlapping.

**Comment by b30wulffz** [comment]
> 1. There are a few errors on the simulation page, which should have been verified before submitting for the review. They are probably coming because the path to load ChemDoodleWeb has not been updated.
> 2. From the experiment it's still not clear what the user is supposed to do with the input and output buttons. This needs to be confirmed with Priya
> 3. On phone resolution, the input and output buttons start overlapping.

Resolved

**Review by raj-vlabs — APPROVED** [review]

**Pull request merged**